### PR TITLE
Use std libs

### PR DESCRIPTION
**Pull request by szymon-filipiak**
Some changes, that improve things. It's still not working correctly but is better xD After this merge I will add the new reader.

### CODE REVIEW

File: header/Text/TextReaderFactory.h [hunk 1]
```diff
@@ -1,17 +1,14 @@
 #ifndef GPU_KARP_RABIN_TEXTREADERFACTORY_H
 #define GPU_KARP_RABIN_TEXTREADERFACTORY_H
 
+#include <bits/unique_ptr.h>
```
Comment (WiktorPieklik) [inline header/Text/TextReaderFactory.h]: `unique_ptr` isn't part of `std` lib? I mean this `include` statement seems to be redundant. If you mind explain it 😄

**Review by WiktorPieklik — COMMENTED** [review]
Everything seems to be ok, but I can no longer compile and run it on macOS. For now I don't know how to fix that. I've got some libraries issues. I hope that you're 100% sure what you're doing as a c++ master 🥇 
I'll very grateful if you answer my comment 👼

**Comment by szymon-filipiak** [comment]
@WiktorPieklik can you check if the errors on your machine are linked to filesystem lib? For now it's a part of cpp 20 standard, but must be additionaly linked in linking process. Maby it has some other name on Mac OS.

File: header/Text/TextReaderFactory.h (hunk 1 above)
Comment (szymon-filipiak) [reply header/Text/TextReaderFactory.h]: Yeah, it is a part of standard library, but is declared in header `memory` or `bits/unique_ptr.h`. Actualy using memory here is cleaner I think, as the documentation states that unique_ptr is declared there. https://en.cppreference.com/w/cpp/memory/unique_ptr

Comment (szymon-filipiak) [reply header/Text/TextReaderFactory.h]: It's the same as with the `std::vector` class. You must do `#include <vector>` to have access to it

**Comment by szymon-filipiak** [comment]
Ok I did some research on filesystem in mac and it looks like apple is slow with the release of this part of the standard in Clang. We should check if there is a clean way for you to use this lib and if not, then decide if we want to use it.

File: header/Utility/TextSplitter.h [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
 private:
     static const int numOfThreads = 32;
 
-    size_t numOfWindows(size_t textLength, int patternLength);
-    std::pair<size_t, size_t> lenOfChains(size_t numOfWindows);
+    static size_t numOfWindows(size_t textLength, size_t patternLength);
+    static std::pair<size_t, size_t> lenOfChains(size_t numOfWindows);
 public:
-    std::vector<std::pair<size_t, size_t>> splitText(long long int textLength, int patternLength);
+    [[maybe_unused]] static std::vector<std::pair<size_t, size_t>> splitText(size_t textLength, size_t patternLength);
```
Comment (zawadzki-dawid) [inline header/Utility/TextSplitter.h:14]: Great use of `maube_unused` attribute, I haven't seen that never before, but it's great :100:

File: main.cpp [hunk 3]
```diff
@@ -8,9 +8,10 @@ int main()
     try
     {
         auto textProcessor = RabinKarpSearch<StandardHash>("../test.txt", "ipsum");
-        std::vector<int> matches = textProcessor.search();
+        std::vector<size_t> matches = textProcessor.search();
 
-        for (const int &i : matches) {
+
+        for (auto i : matches) {
```
Comment (zawadzki-dawid) [inline main.cpp:14]: `auto` looks much cleaner, but I like to stick with static typing, isn't `int` a good call too? It's up to you, but I'm curious why it was chosen :smile:

File: src/Utility/TextSplitter.cpp [hunk 4]
```diff
@@ -1,35 +1,35 @@
...
-size_t TextSplitter::numOfWindows(size_t textLength, int patternLength) {
+size_t TextSplitter::numOfWindows(size_t textLength, size_t patternLength) {
     return textLength - patternLength + 1;
 }
 
 std::pair<size_t, size_t> TextSplitter::lenOfChains(size_t numOfWindows) {
-    int rangeWidth = numOfWindows / numOfThreads;
-    int rangesWithBonus = numOfWindows % numOfThreads;
+    auto rangeWidth = numOfWindows / numOfThreads;
+    auto rangesWithBonus = numOfWindows % numOfThreads;
```
Comment (zawadzki-dawid) [inline src/Utility/TextSplitter.cpp:9]: Same question here :smile:

**Review by zawadzki-dawid — APPROVED** [review]
Code looks great, It can be merged for me, I left some comments for curiosity :smile:. Great Job! :100:

File: main.cpp (hunk 3 above)
Comment (szymon-filipiak) [reply main.cpp:14]: Auto is almost always used with for each loops in cpp. Additionally using reference for simple datatypes as int is not a good practice. It's better to copy them.

File: src/Utility/TextSplitter.cpp (hunk 4 above)
Comment (szymon-filipiak) [reply src/Utility/TextSplitter.cpp:9]: Here if you type int you are doing a conversion from unsigned to signed datatype. Auto will resolve in this case to size_t.

**Review by WiktorPieklik — APPROVED** [review]

**Pull request merged**